From 2ec5aab8252bd3dbfa5712d01f04766bdbab3135 Mon Sep 17 00:00:00 2001
From: Sunil Mohan Adapa <address@hidden>
Date: Mon, 3 Apr 2017 14:31:43 +0530
Subject: [PATCH] parted: Don't warn partition busy in script mode

Script mode promises not to ask for input interactively.  However,
during 'rm' and 'resizepart' operations, then a partition is mounted,
parted asks for confirmation on the terminal.  Since upon confirmation
parted allows the actual operations to proceed allow scripts to do
this as well by assuming 'yes' in both the cases.

Fixes the following issues:
 - #22167
 - #24215
 - lpbug #1270203

Signed-off-by: Sunil Mohan Adapa <address@hidden>
---
 parted/parted.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parted/parted.c b/parted/parted.c
index e84e66d..407ac00 100644
--- a/parted/parted.c
+++ b/parted/parted.c
@@ -231,7 +231,7 @@ _partition_warn_busy (PedPartition* part)
                             PED_EXCEPTION_YES_NO,
                             _("Partition %s is being used. Are you sure you " \
                               "want to continue?"),
-                            path) != PED_EXCEPTION_YES)
+                            path) != PED_EXCEPTION_YES && !opt_script_mode)
                 {
                         free (path);
                         return 0;
-- 
2.17.1

